File: src/Web/ClientApp/src/models/order/Order.ts
export class Order {
  constructor(
    public id?: number,
    public date?: string,
    public comment?: string,
    public status?: DeliveryStatus,
    public buyerId?: number,
    public addressId?: number) {
  }
}

export enum DeliveryStatus {
  Statement,
  InTransit,
  Delivered,
  Canceled
}

File: src/Core/Entities/Order/Order.cs
using System;
using System.Collections.Generic;

namespace Core.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public DateTimeOffset? Date { get; set; }
        public string Comment { get; set; }
        public DeliveryStatus Status { get; set; }
        public int BuyerId { get; set; }
        public int AddressId { get; set; }
        public Address Address { get; set; }
        public Buyer Buyer { get; set; }
        public ICollection<OrderProduct> OrderProducts { get; set; }
        public Order() { }
        public Order(DateTimeOffset? date, string comment, DeliveryStatus status, int buyerId, int addressId)
        {
            Date = date;
            Comment = comment;
            Status = status;
            BuyerId = buyerId;
            AddressId = addressId;
        }
    }

    public enum DeliveryStatus
    {
        Statement,
        InTransit,
        Delivered,
        Canceled
    }
}
